### DISCUSSION TITLE
Calculating gradients for optimization?

### DISCUSSION THREAD

**Discussion by christianhauschel** [byuflowlab/FLOWUnsteady#143] [General]
Are there any plans to make the code capable of computing its gradients "in a smart way" (for gradient-based optimization)?

**Comment by EdoAlvarezR** [comment]
We have been working on an unsteady adjoint solver for the VPM for quite a while now, but it might be a couple of years before that feature gets released openly.

**Comment by EdoAlvarezR** [comment]
It's mostly because the unsteady adjoint requires changing to whole data structure of the particle field, so it's a research code for now.

**Comment by christianhauschel** [comment]
Thank you for the answer. Good to know...
